dvc/command/dag.py

```diff
@@ -30,80 +29,91 @@ def _show_dot(G):
     return dot_file.getvalue()
 
 
-def _build(G, target=None, full=False, outs=False):
+def _collect_targets(repo, target, outs):
+    if not target:
+        return []
+
+    pairs = repo.collect_granular(target)
+    if not outs:
+        return [stage.addressing for stage, _ in pairs]
+
+    targets = []
+    for stage, info in pairs:
+        if not info:
+            targets.extend([str(out) for out in stage.outs])
+            continue
+
+        for out in repo.outs_trie.itervalues(prefix=info.parts):  # noqa: B301
+            targets.extend(str(out))
+
+    return targets
+
+
+def _transform(repo, outs):
     import networkx as nx
 
-    from dvc.repo.graph import get_pipeline, get_pipelines
+    if outs:
+        G = repo.outs_graph
+
+        def _relabel(out):
+            return str(out)
 
-    if target:
-        H = get_pipeline(get_pipelines(G), target)
-        if not full:
-            descendants = nx.descendants(G, target)
-            descendants.add(target)
-            H.remove_nodes_from(set(G.nodes()) - descendants)
     else:
-        H = G
+        G = repo.graph
 
-    if outs:
-        G = nx.DiGraph()
-        for stage in H.nodes:
-            G.add_nodes_from(stage.outs)
+        def _relabel(stage):
+            return stage.addressing
```

Comment:
This whole `_transform` method is only here because `str(stage)` is not using `stage.addressing`. Need to consider changing `Stage.__str__` to it.